src/main/java/com/learn/orderit/actions/UpdateUserAction.java

```diff
@@ -0,0 +1,40 @@
+package com.learn.orderit.actions;
+
+import com.learn.orderit.entity.User;
+import com.learn.orderit.models.UserRequest;
+import com.learn.orderit.repository.IUserRepository;
+import lombok.RequiredArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.Optional;
+
+@Component
+@RequiredArgsConstructor
+public class UpdateUserAction {
+
+    private static final Logger log = LoggerFactory.getLogger(GetUserAction.class);
+    private final IUserRepository userRepository;
+
+    public void invoke(int userId, UserRequest userRequest){
+        log.info("Attempt to update user details");
+        Optional<User> user = userRepository.findById(userId);
+        Optional.ofNullable(user).map(givenUser -> {
```

Comment:
PUT generally replaces an entire object. Ideally, you should convert userRequest to user Entity and persist in the DB.

Field by field updates happens through a PATCH request.